### PR TITLE
Add hidden visibility to protobuf build when we do custom protobuf

### PR DESCRIPTION
**Pull request by Yangqing**
(continuation of #2306 )

### CODE REVIEW

File: cmake/ProtoBuf.cmake [hunk 1]
```diff
@@ -160,6 +174,10 @@ function(caffe2_protobuf_generate_cpp_py srcs_var hdrs_var python_var)
       COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_CURRENT_BINARY_DIR}"
       COMMAND ${CAFFE2_PROTOC_EXECUTABLE} -I${PROJECT_SOURCE_DIR} --cpp_out=${DLLEXPORT_STR}${PROJECT_BINARY_DIR} ${abs_fil}
       COMMAND ${CAFFE2_PROTOC_EXECUTABLE} -I${PROJECT_SOURCE_DIR} --python_out "${PROJECT_BINARY_DIR}" ${abs_fil}
+      COMMAND sed -i "s/::google::protobuf::internal::GetEmptyStringAlreadyInited/GetEmptyStringAlreadyInited/" "${CMAKE_CURRENT_BINARY_DIR}/${fil_we}.pb.h"
```
Comment (yinghai) [inline cmake/ProtoBuf.cmake:177]: Do we have `sed` in windows?

Comment (Yangqing) [reply cmake/ProtoBuf.cmake:177]: Actually not, so @orionr is taking a shot over there. We will probably put windows at a lower priority.

Comment (orionr) [reply cmake/ProtoBuf.cmake:177]: Yup - already got that figured out at https://github.com/caffe2/caffe2/pull/2384 . Unfortunately other things are failing. :) But, yes, Windows will be a lower priority for static linking.

**Comment by orionr** [comment]
This is now rolled into https://github.com/caffe2/caffe2/pull/2384

**Pull request closed without merging**